### PR TITLE
DOCS: a new page for basic usage

### LINKED ISSUE #345: API Docs: add a getting started page
We decided it would be useful to have a page dedicated to a small, geoprocessing workflow that explains some of the most fundamental pieces of pygeoprocessing, such as,

* `get_raster_info` / `get_vector_info`
* `align_and_resize_raster_stack`
* the path-band tuple construct

### PR DESCRIPTION
**Pull request by davemfish**
Fixes #345 

https://pygeoprocessing--349.org.readthedocs.build/en/349/

### CODE REVIEW

File: docs/source/basic_usage.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+.. code::
+
+    import pygeoprocessing
+
+.. code::
+
+    raster_a_path = 'dem_utm.tif'
+    raster_b_path = 'mswep.tif'
+    raster_info = pygeoprocessing.get_raster_info(raster_a_path)
+    raster_info
```
Comment (emlys) [inline docs/source/basic_usage.rst]: Suggest adding `>>> ` to the start of each line to resemble the python REPL and make the distiction between the code and the output below

File: docs/source/basic_usage.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  :func:`raster_calculator <pygeoprocessing.raster_calculator>`.
+
+.. code::
+
+    base_raster_list = [raster_a_path, raster_b_path]
+    target_raster_list = [x.replace('.tif', '_aligned.tif') for x in base_raster_list]
+
+.. code::
+
+    pygeoprocessing.align_and_resize_raster_stack(
```
Comment (emlys) [inline docs/source/basic_usage.rst]: Suggest adding `>>> ` before each line

File: docs/source/basic_usage.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        aggregate_layer_name='watersheds'  # if the vector only contains 1 layer, this can be `None`, or ommitted
+    )
+
+| An example of the path-band object when a function operates on multiple
+  rasters, such as for
+  :func:`raster_calculator <pygeoprocessing.raster_calculator>`:
+
+.. code::
+
+    raster_path_band_list = [(raster_a_path, 1), (raster_b_path, 1)]
```
Comment (emlys) [inline docs/source/basic_usage.rst]: Suggest adding `>>> ` before each line

File: docs/source/basic_usage.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
+Basic Usage
+===========
+
+``pygeoprocessing`` uses GDAL to read and write all GDAL-supported file
+formats. The utility functions
```
Comment (emlys) [inline docs/source/basic_usage.rst]: I've found the GDAL [raster data model](https://gdal.org/user/raster_data_model.html) and [vector data model](https://gdal.org/user/vector_data_model.html) pages to be helpful in understanding the basic concepts. Maybe link to those somewhere in this section?

**Review by emlys — COMMENTED** [review]
Thanks for putting this together @davemfish! I think it will be really helpful to new users. Just a few suggestions

**Comment by davemfish** [comment]
Thanks, @emlys , good suggestions. I made all these changes. The new build is here: https://pygeoprocessing--349.org.readthedocs.build/en/349/basic_usage.html

**Review by emlys — APPROVED** [review]

**Pull request merged**